### DISCUSSION TITLE
src/pages only create a page for lowercased .js files

### DISCUSSION THREAD

**Discussion by fostimus** [gatsbyjs/gatsby#33962] [Ideas / Feature Requests]
Oftentimes we create one off components in the pages directory alongside the page for better organization. They are not reusable in their own right (if they were, they would be in `src/components`), but they help for readability and organization.

The downside is that pages are created for ANY .js file within `src/pages`. This unnecessarily increases build time, and it'd be nice to avoid it with the convention of lowercased .js files being the only ones to actually generate a page.

**Comment by herecydev** [comment]
There's no reason why that one off component couldn't live within your "components" folder. Nothing about components stipulates that they _have_ to be reused. That's not true for "pages" which clearly indicates it's intention. Having any escape hatch convention for pages is going to lead to confusion.

If you really don't like putting those files in components then why not create another folder and reference that? "/one-off-components"

**Reply by fostimus** [reply]
The components that live alongside page .js files make sense to stay there, if they are only used in that one place. `src/components` blows up over the course of a project, and the primary intent for components is to be used in multiple places. I'd rather not put things there that have no intention for reuse.

What I'm proposing here is more about domain setting; if a component only exists for a specific page, it makes sense to keep it alongside that page. If a component is intended for reuse, it makes sense to keep it in a location all developers recognize for reuse (`src/components`).

**Comment by LekoArts** [comment]
Hi! We won't be changing our convention here, files inside `src/pages` are pages and everything else should go somewhere else. Plus, I always use dash-case, lowercased components so e.g. `input-form.js`. Your suggestion would already fall short here.

**Reply by fostimus** [reply]
Curious why you use dash-cased, lowercase names when uppercase and camel case is the convention for React? AFAIK, you can't do any lower casing for components in JSX.

**Reply by LekoArts** [reply]
Sure, React has that convention but the file can be named in which ever way you like. Lowercased ensures no weird uppercase/lowercase OS inconsistency, dash-cased is more readable IMO
